ext/src/http/client/curl/http_operation_curl.cc

```diff
@@ -410,9 +412,99 @@ CURLcode HttpOperation::Setup()
+    }
+    else if (!ssl_options_.ssl_ca_cert_string.empty())
+    {
+#if LIBCURL_VERSION_NUM >= 0x074D00
+      const char *data = ssl_options_.ssl_ca_cert_string.c_str();
+      size_t data_len  = ssl_options_.ssl_ca_cert_string.length();
+
+      struct curl_blob stblob;
+      stblob.data  = const_cast<char *>(data);
+      stblob.len   = data_len;
+      stblob.flags = CURL_BLOB_COPY;
+      curl_easy_setopt(curl_resource_.easy_handle, CURLOPT_CAINFO_BLOB, &stblob);
+      // Assuming PEM format.
+#else
+      // CURL 7.77.0 (0x07 0x4D 0x00) required for CURLOPT_CAINFO_BLOB.
+      return CURLE_UNKNOWN_OPTION;
+#endif
+    }
+
+    /* 2 - CLIENT KEY */
+
+    if (!ssl_options_.ssl_client_key_path.empty())
+    {
+      const char *path = ssl_options_.ssl_client_key_path.c_str();
+      curl_easy_setopt(curl_resource_.easy_handle, CURLOPT_SSLKEY, path);
+      curl_easy_setopt(curl_resource_.easy_handle, CURLOPT_SSLKEYTYPE, "PEM");
+    }
+    else if (!ssl_options_.ssl_client_key_string.empty())
+    {
+#if LIBCURL_VERSION_NUM >= 0x074700
+      const char *data = ssl_options_.ssl_client_key_string.c_str();
+      size_t data_len  = ssl_options_.ssl_client_key_string.length();
+
+      struct curl_blob stblob;
+      stblob.data  = const_cast<char *>(data);
+      stblob.len   = data_len;
+      stblob.flags = CURL_BLOB_COPY;
+      curl_easy_setopt(curl_resource_.easy_handle, CURLOPT_SSLKEY_BLOB, &stblob);
+      curl_easy_setopt(curl_resource_.easy_handle, CURLOPT_SSLKEYTYPE, "PEM");
+#else
+      // CURL 7.71.0 (0x07 0x47 0x00) required for CURLOPT_SSLKEY_BLOB.
+      return CURLE_UNKNOWN_OPTION;
+#endif
+    }
+
+    /* 3 - CLIENT CERT */
+
+    if (!ssl_options_.ssl_client_cert_path.empty())
+    {
+      const char *path = ssl_options_.ssl_client_cert_path.c_str();
+      curl_easy_setopt(curl_resource_.easy_handle, CURLOPT_SSLCERT, path);
+      curl_easy_setopt(curl_resource_.easy_handle, CURLOPT_SSLCERTTYPE, "PEM");
+    }
+    else if (!ssl_options_.ssl_client_cert_string.empty())
+    {
+#if LIBCURL_VERSION_NUM >= 0x074700
+      const char *data = ssl_options_.ssl_client_cert_string.c_str();
+      size_t data_len  = ssl_options_.ssl_client_cert_string.length();
+
+      struct curl_blob stblob;
+      stblob.data  = const_cast<char *>(data);
+      stblob.len   = data_len;
+      stblob.flags = CURL_BLOB_COPY;
+      curl_easy_setopt(curl_resource_.easy_handle, CURLOPT_SSLCERT_BLOB, &stblob);
+      curl_easy_setopt(curl_resource_.easy_handle, CURLOPT_SSLCERTTYPE, "PEM");
+#else
+      // CURL 7.71.0 (0x07 0x47 0x00) required for CURLOPT_SSLCERT_BLOB.
+      return CURLE_UNKNOWN_OPTION;
+#endif
+    }
+
+    /* 4 - ENFORCE VERIFICATION */
+
+    curl_easy_setopt(curl_resource_.easy_handle, CURLOPT_USE_SSL, (long)CURLUSESSL_ALL);
+
+    // FIXME: Need to get a real CERT, not self signed, for testing:
+    curl_easy_setopt(curl_resource_.easy_handle, CURLOPT_SSL_VERIFYPEER, 1L);
```

Comment:
Thanks @owent

On one hand, we need an option to relax the code to use `CURLOPT_SSL_VERIFYPEER = 0L` to run unit tests with a self signed certificate.

On the other hand, disabling `CURLOPT_SSL_VERIFYPEER` defeats the purpose of `SSL`, and is not secure, so this option should never leak into production.

My concern is to find a way to satisfy the former for testing, without creating a hole with the later.

If we have an option for that, it should be abundantly clear that it is for testing only, not to be used in production. Maybe it can be available in debug releases only. I will investigate.